src/DefaultBuilder/test/Microsoft.AspNetCore.Tests/WebApplicationTests.cs

```diff
@@ -600,26 +601,36 @@ public void WebApplicationBuilderCanModifyWebRootAfterCreateBuilder()
             builder.WebHost.UseSetting("WEBROOT", webRoot);
 
             Assert.Equal(webRoot, builder.WebHost.GetSetting("webroot"));
+            Assert.IsType<NullFileProvider>(builder.Environment.WebRootFileProvider);
 
             var app = builder.Build();
 
             Assert.Equal(webRoot, app.Environment.WebRootPath);
+            Assert.Equal(webRoot, ((PhysicalFileProvider)app.Environment.WebRootFileProvider).Root);
         }
 
         [Fact]
         public void WebApplicationBuilderWebRootIsRelativeToContentRoot()
         {
-            var contentRoot = Path.GetTempPath().ToString();
+            // We use the current test directory since it is guranteed not
+            // to have a wwwroot directory by default which will prevent the default
+            // initialization logic in HostingEnvironmentExtensions.Initialize from
+            // kicking in and setting the WebRootFileProvider to {ContentRoot}/wwwroot
+            var contentRoot = System.IO.Directory.GetCurrentDirectory();
```

Comment:
Maybe just create and delete a directory at `Path.GetTempFileName()` instead.

Reply:
Yeah, admittedly, assuming that the test directory will never have a `wwwroot` isn't the best assumption but I was optimizing for not having an I/O in the tests. I'll change it though since I don't have a strong opinion either way.